src/core/tests/aio.cpp

```diff
@@ -34,6 +34,7 @@
  */
 
 
+# include <future>
```

Comment:
best to not use future as it is too new to be supported. + it introduces some hidden dependnecies